src/main/kotlin/uk/gov/dwp/dataworks/aws/AwsCommunicator.kt

```diff
@@ -266,13 +327,55 @@ class AwsCommunicator {
         return role
     }
 
+    /**
+     * Deletes the [Role] with the specified [roleName]
+     */
+    fun deleteIamRole(correlationId: String, roleName: String) {
+        awsClients.iamClient.deleteRole(DeleteRoleRequest.builder().roleName(roleName).build())
+        logger.info("Deleted iam role", "correlation_id" to correlationId, "role_name" to roleName)
+    }
+
     /**
      * Attaches [policy] to [role] using the [Policy] ARN and [Role] name.
      */
-    fun attachIamPolicyToRole(policy: Policy, role: Role) {
+    fun attachIamPolicyToRole(correlationId: String, policy: Policy, role: Role) {
         awsClients.iamClient.attachRolePolicy(AttachRolePolicyRequest.builder()
                 .policyArn(policy.arn())
                 .roleName(role.roleName()).build())
-        logger.info("Attched policy to role", "policy_arn" to policy.arn(), "role_name" to role.roleName())
+        logger.info("Attched policy to role", "correlation_id" to correlationId, "policy_arn" to policy.arn(), "role_name" to role.roleName())
+    }
+
+    fun createDynamoDbTable(tableName: String, attributes: List<AttributeDefinition>, keyName: String) {
```

Comment:
What happens if the table already exists? i.e when we restart the service?

Reply:
Yep, I'll wrap this into a check first